description: 'cipso,calipso: resolve a number of problems with the DOI refcounts'
references:
- https://sites.google.com/view/syzscope/kasan-use-after-free-read-in-cipso_v4_genopt
- https://syzkaller.appspot.com/bug?id=96e7d345748d8814901c91cd92084ed04b46701e
- https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2021-33033
- https://git.kernel.org/linus/ad5d07f4a9cd671233ae20983848874731102c08
- https://git.kernel.org/linus/1165affd484889d4986cf3b724318935a0b120d8
- https://cdn.kernel.org/pub/linux/kernel/v5.x/ChangeLog-5.11.14
- https://cdn.kernel.org/pub/linux/kernel/v5.x/ChangeLog-5.11.7
comments:
  debian/bwh: |-
    The "second commit" in ieee802154 (1165affd4848) is fixing a
    totally different issue.
  debian/carnil: |-
    First commit required landed in 4.19.181, 5.10.24, 5.12-rc3.
    Second one in 4.19.187, 5.10.30, 5.12-rc7.
  ubuntu/sbeattie: |-
    looks to be introduced primarily in d7cce01504a0
    1165affd484889d4986cf3b724318935a0b120d8 seems like an unrelated
      separate issue?
  wens: Introduced in v2.6.28-rc1 and v4.8-rc1; fixed in v5.12-rc3.
introduced-by:
  mainline: [b1edeb102397546438ab4624489c6ccd7b410d97, d7cce01504a0ccb95b5007d846560cfccbc1947f]
fixed-by:
  cip/4.19: [a44af1c69737f9e64d5134c34eb9d5c4c2e04da1]
  cip/4.19-rt: [a44af1c69737f9e64d5134c34eb9d5c4c2e04da1]
  debian/buster-security: ['version:debian/4.19.181-1']
  mainline: [ad5d07f4a9cd671233ae20983848874731102c08]
  stable/4.14: [ab44f7317c16ddcf9ee12ba2aca60771266c2dc6]
  stable/4.19: [a44af1c69737f9e64d5134c34eb9d5c4c2e04da1]
  stable/5.10: [85178d76febd30a745b7d947dbd9751919d0fa5b]
  stable/5.11: [00d566df2cceb8591913b3ea3b43d2918915f7e3]
  stable/5.4: [b4800e7a1c9f80a1a0e417ab36a1da4959f8b399]
  ubuntu/bionic: [c2862f4b448bbd2492427ee7ad0530a640735ce0]
  ubuntu/focal: [893e092f599947a87f6634799abf668d405a58fa]
  ubuntu/groovy: [d0e918b8f422d2c5b9970beaf5ef4b8e46b0995a]
